tor/controller.go

```diff
@@ -141,9 +145,15 @@ type Controller struct {
 // a Tor server.
 func NewController(controlAddr string, targetIPAddress string,
 	password string) *Controller {
-
+	cleanedControlAddr := controlAddr
+	controlAddrType := "tcp"
+	if strings.HasPrefix(controlAddr, "unix://") {
+		controlAddrType = "unix"
+                cleanedControlAddr = strings.Replace(controlAddr, "unix://", "", -1)
```

Comment:
nit: use `ReplaceAll` to remove the need for the `-1` parameter.

Reply:
Fixed